read-file.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+type Node struct {
+	*MountInfoLine
+	Children []*Node `json:"children,omitempty"`
+}
+
+type MountInfoLine struct {
+	MountID        string   `json:"-"`
+	ParentID       string   `json:"-"`
+	MajorMinor     string   `json:"-"`
+	Root           string   `json:"-"`
+	MountPoint     string   `json:"name"`
+	MountOptions   string   `json:"-"`
+	OptionalFields []string `json:"-"`
+	FileSystemType string   `json:"-"`
+	MountSource    string   `json:"-"`
+	SuperOptions   string   `json:"-"`
+}
+
+func extractMountinfo(flp string) ([]*MountInfoLine, error) {
+	var mi []*MountInfoLine
+
+	mifl, err := os.Open(flp)
+	if err != nil {
+		return nil, err
+	}
+	defer mifl.Close()
+
+	sc := bufio.NewScanner(mifl)
+	for sc.Scan() {
+		line := strings.Split(sc.Text(), " ")
+		lineLen := len(line)
+		if lineLen < 9 {
+			return nil, fmt.Errorf("not enough fields in the mountinfo file: %v", line)
+		}
+		mountinfo := &MountInfoLine{
+			// TODO: find a cleaner way of doing this
+			MountID:        line[0],
+			ParentID:       line[1],
+			MajorMinor:     line[2],
+			Root:           line[3],
+			MountPoint:     line[4],
+			MountOptions:   line[5],
+			FileSystemType: line[(lineLen-10)+7],
```

Comment:
yes will look at a better implementation. Right now I'm calculating based on the number of optional fields and its not the most reliable I guess.